Use Closure's debug loader

schmidtk opened the pull request:
Closure provides its own debug loader in `base.js` that handles loading both legacy `goog.provide` scripts and `goog.module` scripts, based on a generated dependency list. This PR adds a new `writeDebugLoader` call to generate a script containing:

- `goog.addDependency` calls for all source files, generated with [Closure's DepsWriter](https://developers.google.com/closure/library/docs/depswriter).
- A call to ensure `goog.module` files wait on legacy `goog.provide` dependencies to load before loading themselves.
- A `goog.bootstrap` call to load the application entry point(s).

Other changes:
- Abstract code to run a Python command and return its output into `execPythonCmd`.
- Create utility functions to clean up filter/map/etc calls.

Commit:
feat(debug): create an app loader that uses the gcc debug loader

Commit:
refactor: reuse python runner and clean up code

wallw-dev commented on .eslintrc.js:
```diff
@@ -14,7 +14,7 @@ module.exports = {
     "no-negated-condition": "off",
     // allow use of var
-    "no-var": "off",
+    "no-var": "error",
```
If this is changing, shouldn't the comment change as well?

schmidtk replied:
Yes, thanks. Removed it because it's already an error in `eslint-config-google`.

Commit:
fix(goog): reject promise if closure scripts cannot be found

BREAKING CHANGE: Builds will now fail if they attempt to run closurebuilder.py
or depswriter.py and the script cannot be found.

Commit:
chore(eslint): use no-var rule from eslint-config-google

wallw-dev commented:
:tada: This PR is included in version 4.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/opensphere-build-closure-helper)
- [GitHub release](https://github.com/ngageoint/opensphere-build-closure-helper/releases/tag/v4.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
